lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -154,6 +154,43 @@ printMemoryAccessAttribute(LoadStoreOpTy loadStoreOp, OpAsmPrinter &printer,
   elidedAttrs.push_back(spirv::attributeName<spirv::StorageClass>());
 }
 
+template <typename CastOp>
+static LogicalResult verifyCastOp(CastOp castOp,
+                                  bool sameOperandAndResultBitWidth = true) {
+  Type operandType = castOp.operand()->getType();
+  Type resultType = castOp.result()->getType();
+
+  // ODS checks that result type and operand type have the same shape.
+  if (auto vectorType = operandType.dyn_cast<VectorType>()) {
+    operandType = vectorType.getElementType();
+    resultType = resultType.cast<VectorType>().getElementType();
+  }
+
+  const auto operandTypeBitWidth = operandType.getIntOrFloatBitWidth();
+  const auto resultTypeBitWidth = resultType.getIntOrFloatBitWidth();
+  const auto isSameBitWidth = operandTypeBitWidth == resultTypeBitWidth;
+
+  if (sameOperandAndResultBitWidth) {
```

Comment:
I don't think the spec requires `OpConvertFToS`/etc. to be of the same bitwidth?

Reply:
@antiagainst thanks for review, I was looking into DirectXShaderCompiler for this ops https://github.com/microsoft/DirectXShaderCompiler/blob/master/tools/clang/lib/SPIRV/SpirvEmitter.cpp#L6623 , as far as I understood those code, before creating an instruction `OpConvertFToS` it explicitly converts a bit width of `fromType` into bit width of `toType`  by `OpFConvert` instruction.

> // First convert the source to the bitwidth of the destination if necessary.

please fix me if I'm wrong.
Thanks.

Reply:
The spec itself does not seem to have this restriction. But I think it is better that in the SPIR-V dialect we enforce the bit width to be same for the conversion and use OpFConvert/OpSConvert to model the bitwidth conversions.

Reply:
@denis0x0D: haha, good find! Actually I cannot recall exactly why we did that in DXC... It might be the spec wording changed or something. What Mahesh said makes sense. Fine for me to keep it.